core/templates/dev/head/pages/collection_editor/CollectionEditorNavbarDirective.js

```diff
@@ -68,11 +78,15 @@ oppia.directive('collectionEditorNavbar', [function() {
           // action since it is not reversible.
           CollectionRightsBackendApiService.setCollectionPublic(
             $scope.collectionId, $scope.collection.getVersion()).then(
-            function() {
-              // TODO(bhenning): There should be a scope-level rights object
-              // used, instead. The rights object should be loaded with the
-              // collection.
-              $scope.isPrivate = false;
+            function(data) {
+              if (!data.is_private) {
+                $scope.isPrivate = data.is_private;
+                $rootScope.$broadcast(EVENT_COLLECTION_STATUS_CHANGE);
```

Comment:
I think a better way to do this would be to create a ``CollectionRightsObjectFactory`` to represent a rights domain object for collections, then return it here. Such an object would be cached in subsequent calls to ``get``, and would be locally updated in calls to ``set`` after the backend RPC succeeds. Instead of watching a global event, you could watch the object returned from ``get`` for any changes (which would happen by calls to ``set``). This feels cleaner to me, as it decouples the requirement for updating state in another part of the app from the UI (though ``$scope`` is still needed for ``$watch``).